src/backend_task/identity/register_dpns_name.rs

```diff
@@ -148,6 +149,58 @@ impl AppContext {
             .await
             .map_err(|e| e.to_string())?;
 
-        Ok(())
+        // Re-fetch the identity's DPNS names from Platform
+        let dpns_names_document_query = DocumentQuery {
+            data_contract: self.dpns_contract.clone(),
+            document_type_name: "domain".to_string(),
+            where_clauses: vec![WhereClause {
+                field: "records.identity".to_string(),
+                operator: WhereOperator::Equal,
+                value: Value::Identifier(qualified_identity.identity.id().into()),
+            }],
+            order_by_clauses: vec![],
+            limit: 100,
+            start: None,
+        };
+
+        let owned_dpns_names = Document::fetch_many(&self.sdk, dpns_names_document_query)
+            .await
+            .map(|document_map| {
+                document_map
+                    .values()
+                    .filter_map(|maybe_doc| {
+                        maybe_doc.as_ref().and_then(|doc| {
+                            let name = doc
+                                .get("normalizedLabel")
+                                .map(|label| label.to_str().unwrap_or_default());
+                            let acquired_at = doc
+                                .created_at()
+                                .into_iter()
+                                .chain(doc.transferred_at())
+                                .max();
+
+                            match (name, acquired_at) {
+                                (Some(name), Some(acquired_at)) => Some(DPNSNameInfo {
+                                    name: name.to_string(),
+                                    acquired_at,
+                                }),
+                                _ => None,
+                            }
+                        })
+                    })
+                    .collect::<Vec<DPNSNameInfo>>()
+                    .into()
+            })
+            .map_err(|e| format!("Error fetching DPNS names: {}", e))?;
+
+        qualified_identity.dpns_names = owned_dpns_names;
+
+        // Insert qualified identity into the database
+        self.insert_local_qualified_identity(&qualified_identity)
+            .map_err(|e| format!("Database error: {}", e))?;
+
+        Ok(BackendTaskSuccessResult::Message(
+            "Successfully registered dpns name".to_string(),
+        ))
```

Comment:
Why do this, the put to platform and wait for response gives you the document that you just inserted as proof.

Reply:
(added a to-do because documents are returned for contested names as well)